manifests/init.pp

```diff
@@ -9,6 +9,11 @@
   $tools            = true,
 ) inherits pxe::params {
 
+  # Ensure wget is installed before any execs of wget
+  Package <| |> -> Exec <| |>
+  package { 'wget':
```

Comment:
I think you should use ```ensure_resource()``` instead, to prevent duplicate resource declarations.

Reply:
or more specialized, `ensure_packages`

Reply:
The side affect of this code will be realizing all virtual packages and execs, which is not good